lib/nimble_parsec/compiler.ex

```diff
@@ -893,7 +893,7 @@ defmodule NimbleParsec.Compiler do
 
     prefix =
       cond do
-        :integer in modifiers -> "byte"
+        :integer in modifiers -> "ASCII character"
```

Comment:
Thank you for the PR! I think we need to check if the inclusive and exclusive ranges are all asciis, if so, we use ASCII characters, otherwise we use byte. WDYT?